File: Brian Kernighans algorithm/Brian_Kernighans_algo.go
package main

import "fmt"

func hammingWeight(num uint32) int {
	count := 0
	for num > 0 {
		num = num & (num - 1) // clear the rightmost set bit
		count += 1
	}

	return count
}

func main() {
	number := 22
	fmt.Printf("%d in binary is %b\n", number, number)
	fmt.Printf("the number of set bits in %d is %d\n", number, hammingWeight(22))
}

/*
output:
22 in binary is 10110
the number of set bits in 22 is 3
*/

File: Brian Kernighans algorithm/Brian_Kernighans_algo.py
class Solution(object):
    def hammingWeight(self, n):
        """
        :type n: int
        :rtype: int
        """
        count = 0
        while n:
            n = n & (n-1) # clear the rightmost set bit
            count += 1
            
        return count


if __name__ == "__main__":
    solution = Solution()
    number = 22
    print("{0:d} in binary is {1:b}".format(number, number))
    print("the number of set bits in {0:d} is {1:d}".format(number, solution.hammingWeight(number)))
    
"""
22 in binary is 10110
the number of set bits in 22 is 3
"""